### ISSUE TITLE
Can not start siad after crash on No space left on device

### ISSUE THREAD

**Issue by starius** [NebulousLabs/Sia#2934]
# BUG REPORT

My Sia 1.3.0 crashed with this error message:

```
oroutine 295885706 [running]:
runtime/debug.Stack(0x445657, 0x0, 0xc421cabc80)
        /home/sia/.goroot/src/runtime/debug/stack.go:24 +0x79
runtime/debug.PrintStack()
        /home/sia/.goroot/src/runtime/debug/stack.go:16 +0x22
github.com/NebulousLabs/Sia/build.Severe(0xc421cabe58, 0x2, 0x2)
        /home/sia/src/github.com/NebulousLabs/Sia/build/critical.go:32 +0x92
github.com/NebulousLabs/Sia/persist.(*Logger).Severe(0xc4200c6760, 0xc421cabe58, 0x2, 0x2)
        /home/sia/src/github.com/NebulousLabs/Sia/persist/log.go:71 +0xe0
github.com/NebulousLabs/Sia/modules/host/contractmanager.(*writeAheadLog).commit.func2(0xc4deeaecd0, 0xc420084428, 0x0, 0x0, 0x0, 0xf07908, 0x0, 0x0)
        /home/sia/src/github.com/NebulousLabs/Sia/modules/host/contractmanager/writeaheadlogsync.go:201 +0x357
created by github.com/NebulousLabs/Sia/modules/host/contractmanager.(*writeAheadLog).commit
        /home/sia/src/github.com/NebulousLabs/Sia/modules/host/contractmanager/writeaheadlogsync.go:209 +0x1e5
Severe error: Unable to properly initialize WAL file, crashing to prevent corruption: unable to write WAL metadata: write host/contractmanager/contractmanager.wal_temp: no space left on device

goroutine 295885706 [running]:
runtime/debug.Stack(0x445657, 0x0, 0xc421cabaa0)
        /home/sia/.goroot/src/runtime/debug/stack.go:24 +0x79
runtime/debug.PrintStack()
        /home/sia/.goroot/src/runtime/debug/stack.go:16 +0x22
github.com/NebulousLabs/Sia/build.Severe(0xc421cabc18, 0x2, 0x2)
        /home/sia/src/github.com/NebulousLabs/Sia/build/critical.go:32 +0x92
github.com/NebulousLabs/Sia/persist.(*Logger).Severe(0xc4200c6760, 0xc421cabc18, 0x2, 0x2)
        /home/sia/src/github.com/NebulousLabs/Sia/persist/log.go:71 +0xe0
github.com/NebulousLabs/Sia/modules/host/contractmanager.(*writeAheadLog).appendChange(0xc420084428, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /home/sia/src/github.com/NebulousLabs/Sia/modules/host/contractmanager/writeaheadlog.go:151 +0x46e
github.com/NebulousLabs/Sia/modules/host/contractmanager.(*writeAheadLog).commit.func2(0xc4deeaecd0, 0xc420084428, 0x0, 0x0, 0x0, 0xf07908, 0x0, 0x0)
        /home/sia/src/github.com/NebulousLabs/Sia/modules/host/contractmanager/writeaheadlogsync.go:208 +0x423
created by github.com/NebulousLabs/Sia/modules/host/contractmanager.(*writeAheadLog).commit
        /home/sia/src/github.com/NebulousLabs/Sia/modules/host/contractmanager/writeaheadlogsync.go:209 +0x1e5
Severe error: Unable to write state change to WAL: write host/contractmanager/contractmanager.wal_temp: no space left on device
panic: unable to append a change to the WAL, crashing to prevent corruption

goroutine 295885706 [running]:
github.com/NebulousLabs/Sia/modules/host/contractmanager.(*writeAheadLog).appendChange(0xc420084428, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /home/sia/src/github.com/NebulousLabs/Sia/modules/host/contractmanager/writeaheadlog.go:152 +0x4b8
github.com/NebulousLabs/Sia/modules/host/contractmanager.(*writeAheadLog).commit.func2(0xc4deeaecd0, 0xc420084428, 0x0, 0x0, 0x0, 0xf07908, 0x0, 0x0)
        /home/sia/src/github.com/NebulousLabs/Sia/modules/host/contractmanager/writeaheadlogsync.go:208 +0x423
created by github.com/NebulousLabs/Sia/modules/host/contractmanager.(*writeAheadLog).commit
        /home/sia/src/github.com/NebulousLabs/Sia/modules/host/contractmanager/writeaheadlogsync.go:209 +0x1e5
```

I added free space and tried to start it again and got this in console:

```
$ siad -M gctwhr
Loading...
(0/6) Loading siad...
(1/6) Loading gateway...
(2/6) Loading consensus...
(3/6) Loading transaction pool...
(4/6) Loading wallet...
(5/6) Loading host...
Closing wallet...
Closing transaction pool...
Closing consensus...
Closing gateway...
error while loading contract manager atomic data: error loading the contract manager settings file: unable to read persisted json object from disk: unable to read header from persisted json object file: EOF
```

Content of some files:

```
$ cat host/contractmanager/contractmanager.wal
{
        "Header": "Sia Contract Manager WAL",
        "Version": "1.2.0"
}{
        "ErroredStorageFolderAdditions": null,
        "ErroredStorageFolderExtensions": null,
        "StorageFolderAdditions": null,
        "StorageFolderExtensions": null,
        "StorageFolderRemovals": null,
        "StorageFolderReductions": null,
        "UnfinishedStorageFolderAdditions": null,
        "UnfinishedStorageFolderExtensions": [],
        "SectorUpdates": null
}
$ cat host/contractmanager/contractmanager.wal_temp
$ cat host/contractmanager/contractmanager.json 
$ cat host/contractmanager/contractmanager.json_temp
```

**Expected Behavior**

Successful restart of `siad`

**Environment**
* Sia version: 1.3.0
* OS: Linux

**Comment by DavidVorick** [comment]
pretty sure this is a corruption issue that we fixed in v1.3.1 :/

It looks like the corruption isn't that bad though, you might be able to fix it by hand if you modify the json file

**Comment by starius** [comment]
All json files looks empty except contractmanager.wal which just looks as a default one.

**Comment by DavidVorick** [comment]
That's not great. It means your host is probably not recoverable, you should be able to start siad if you wipe your host directory.

I think this is a bug we already resolved (in v1.3.1), but I'm not 100% sure, so I'm going to leave this open. If anyone else experiences the same issue on a more recent version, please let me know and we'll investigate further.

**Comment by tbenz9** [comment]
This issue was very common with 1.3.0 and was tracked here https://github.com/NebulousLabs/Sia/issues/2407

I'm going to close the issue as we believe it was resolved, and if it's not, we would probably re-open https://github.com/NebulousLabs/Sia/issues/2407 instead of tracking it in this issue.
